### ISSUE TITLE
Support JIRA renderer

### ISSUE THREAD

**Issue by kickingvegas** [miyuchina/mistletoe#10]
Feature request - add support for rendering Markdown (in particular GFM) to JIRA markup.

JIRA markup documentation at this link:
https://jira.atlassian.com/secure/WikiRendererHelpAction.jspa?section=all

**Comment by miyuchina** [comment]
Would you have time to do it yourself? It essentially boils down to writing a custom renderer, for which you can refer to `mistletoe/html_renderer.py` as an example. Here's what, say, the render function for JIRA's headings will look like:

```python
from mistletoe.base_renderer import BaseRenderer

class JIRARenderer(BaseRenderer):
    def render_heading(self, token):
        template = 'h{level}. {inner}\n'
        inner = self.render_inner(token)
        return template.format(level=token.level, inner=inner)
```

The reason I'm not implementing this myself is that I don't think rendering to JIRA markup should be included as a core feature of mistletoe. I'm currently focusing on spec compliance and documentation for mistletoe. If someone implements this (or if I got around to implementing this), though, I'd be happy to include it in the `plugins` directory.

If you feel we should include a JIRA renderer in `plugins`, let me know and I'll mark this issue as "help wanted," so that people can help write up an implementation. Also feel free to email / open other issues if you have any trouble implementing the renderer. I understand that API documentation is currently lacking, but looking at `HTMLRenderer` should give you some good ideas on how to implement it.

**Comment by kickingvegas** [comment]
Perhaps. If I have a couple of hours I can see how far I get with making a JIRA renderer. Probably the biggest unknown are the tags like `level` and `inner` that I'll need to know to make the right mapping.

**Comment by miyuchina** [comment]
Yeah, that's what I thought. I hope to have the API stabilized some time soon so that I can lock down the documentation for those tags. I apologize for this.

But `HTMLRenderer` makes use of all the available tags, so you can look at those. If a `token` has attribute `children`, use `self.render_inner(token)` to render everything inside that token.

**Comment by cctile** [comment]
Started working on this, making decent progress. Found a bug in the table rendering code which tries to iterate a tuple for headers.

**Comment by cctile** [comment]
Got some early work here https://github.com/cctile/mistletoe/tree/cc/jira-support

**Comment by cctile** [comment]
Want to build a standalone script `md2jira.py`; if I add to the PR, can I create a `contrib` directory to put that file there?

**Comment by miyuchina** [comment]
Implemented in #11 . Closing for now.
